src/sentry/static/sentry/app/components/forms/formField.jsx

```diff
@@ -38,18 +38,21 @@ export default class FormField extends React.PureComponent {
   constructor(props, context) {
     super(props, context);
     this.state = {
+      error: null,
       value: this.getValue(props, context),
     };
   }
 
   componentDidMount() {}
 
   componentWillReceiveProps(nextProps, nextContext) {
-    if (
-      this.props.value !== nextProps.value ||
-      (!defined(this.context.form) && defined(nextContext.form))
-    ) {
-      this.setValue(this.getValue(nextProps, nextContext));
+    let newError = this.getError(nextProps, nextContext);
+    let newValue = this.getValue(nextProps, nextContext);
+    if (newError != this.state.error) {
```

Comment:
there's two setState calls here, which kind of is meh, but I **think** it'll be fine

Reply:
It's meh, but it's fine, react batches setState calls.